analytics-services/src/main/java/org/exoplatform/analytics/es/service/ESAnalyticsService.java

```
@@ -0,0 +1,580 @@
+
+  private static final Context               CONTEXT                               = Context.GLOBAL.id("ANALYTICS");
+
+  private static final Scope                 ES_SCOPE                              = Scope.GLOBAL.id("elasticsearch");
+
+  private static final String                ES_AGGREGATED_MAPPING                 = "ES_AGGREGATED_MAPPING";
+
+  private static final AnalyticsFieldFilter  ES_TYPE_FILTER                        =
+                                                            new AnalyticsFieldFilter("isAnalytics",
+                                                                                     AnalyticsFieldFilterType.EQUAL,
+                                                                                     "true");
+
+  private AnalyticsESClient                  esClient;
+
+  private SettingService                     settingService;
+
+  private Map<String, StatisticFieldMapping> esMappings                            = new HashMap<>();
+
+  private ScheduledExecutorService           esMappingUpdater                      = Executors.newScheduledThreadPool(1);
+
+  private String                             administratorsPermission;
+
+  public ESAnalyticsService(SettingService settingService, AnalyticsESClient esClient, InitParams params) {
+    this.esClient = esClient;
+    this.settingService = settingService;
+
+    if (params != null && params.containsKey(ANALYTICS_ADMIN_PERMISSION_PARAM_NAME)) {
+      this.administratorsPermission = params.getValueParam(ANALYTICS_ADMIN_PERMISSION_PARAM_NAME).getValue();
+    }
+  }
+
+  @Override
+  public void start() {
+    // Can't be job, because the mapping retrival must be executed on each
+    // cluster node
+    esMappingUpdater.scheduleAtFixedRate(() -> {
+      try {
+        retrieveMapping(true);
+      } catch (Throwable e) {
+        if (LOG.isDebugEnabled()) {
+          LOG.warn("Error while checking connection status to Etherreum Websocket endpoint", e);
```

The level is not in relation with the if.
You shoud not swallow the exception in this case IMO

---

done